tests/exec/execute.go

```diff
@@ -0,0 +1,88 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package exec
+
+import (
+	"bytes"
+	"fmt"
+
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/remotecommand"
+	"kubevirt.io/client-go/kubecli"
+)
+
+func ExecuteCommandOnPod(virtCli kubecli.KubevirtClient, pod *k8sv1.Pod, containerName string, command []string) (string, error) {
+	stdout, stderr, err := ExecuteCommandOnPodV2(virtCli, pod, containerName, command)
+
+	if err != nil {
+		return "", fmt.Errorf("failed executing command on pod: %v: stderr %v: stdout: %v", err, stderr, stdout)
+	}
+
+	if len(stderr) > 0 {
+		return "", fmt.Errorf("stderr: %v", stderr)
+	}
+
+	return stdout, nil
+}
+
+func ExecuteCommandOnPodV2(virtCli kubecli.KubevirtClient, pod *k8sv1.Pod, containerName string, command []string) (stdout, stderr string, err error) {
+	var (
+		stdoutBuf bytes.Buffer
+		stderrBuf bytes.Buffer
+	)
+	options := remotecommand.StreamOptions{
+		Stdout: &stdoutBuf,
+		Stderr: &stderrBuf,
+		Tty:    false,
+	}
+	err = ExecCommandOnPod(virtCli, pod, containerName, command, options)
+	return stdoutBuf.String(), stderrBuf.String(), err
+}
+
+func ExecCommandOnPod(virtCli kubecli.KubevirtClient, pod *k8sv1.Pod, containerName string, command []string, options remotecommand.StreamOptions) error {
```

Comment:
@ShellyKa13 Hi, yes, for example this one can be changed to `ExecuteCommandOnPodWithOptions`. I don't know what was the motivation for having `ExecuteCommandOnPod` and `ExecuteCommandOnPodV2`, but from what I see they have different users.

Reply:
right exactly I think its better to have unique names. So if you are moving them changing the name at the same opportunity will be nice. Ping me when you do that and ill lgtm

Reply:
Done in https://github.com/kubevirt/kubevirt/pull/8454/commits/a332fb9d67c2e958835dd141129b9c28b54f0c17. Thank you :)